### PR TITLE
[NVIDIA TF] Relax protobuf version requirement

### PR DESCRIPTION
**Pull request by nluehr**
Relax protobuf python package requirement to allow 3.x versions above 3.20.
This brings the requirement into line with other packages such as tensorboard.

### CODE REVIEW

**Comment by nluehr** [comment]
I'm not sure about Windows. We could be conservative and do something like
```
'protofbuf >= 3.9.2, < 3.20; platform_system=="Windows"`
'protofbuf >= 3.9.2, < 4; platform_system!="Windows"`
```

**Comment by vam-google** [comment]
tl;dr; it is a very good idea to have 3.20 accepted by TF, as we are soon to require protobuf >= 3.20, but a proper testing (and upgrading our CI) is required before pushing this, unless it already has been done. More details below.


How urgent is this PR? In principle the PR looks fine and we are about to push updates for protobuf to allow even protobuf 4 (and actually disallow 3.19 and below). 

The description in the comment (https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py#L100) says that 3.20 causes linker errors but it is not clear which specific combination of protobuf runtime and generation time dependencies were causing that and I can't reproduce that with 3.20 runtime and current migration to protobuf 4 i'm working on now. 

We will be soon rolling out TF with `protobuf >= 3.20, <5` restrictions, since 3.20 is the only officially supported version of protobuf runtime compatible with both protoc 3 and protoc 4 (and we are migrating to protoc 4 to support many other important customers like Google Ads and Google Cloud). 

Having an intermediate variant with `protobuf >= 3.9, <= 3.20` right before rolling out `protobuf >= 3.20 < 5` seems like a good idea, as it gives us an intersecting protobuf version compatible with both old ad future (protobuf-4) TF.

The only issue is we need to have it properly tested on protobuf 3.20 runtime 3.9.2 generationtime on windows. Currently our CI and corresponding docker images still install/depend on protobuf 3.19 runtime, so even if we release the requirement here it most likely will not affect our CI and all tests will pass, because in reality it will be still testing it on 3.19.

**Comment by groodt** [comment]
If this PR was merged, which tensorflow version would it be targetting? Or will tensorflow 2.12 be targetting newer protobuf version?

**Comment by vam-google** [comment]
@groodt I believe next version of TF release will be targeting protobuf 4, as today I finally got it all build and test pass on Windows with protoc-3.21.9 (corresponds to 4.21.9 in python). Specifically, given the compatibility constraints, we will have to specify tf pypi package to be constrained as `protobuf >= 3.20 < 5`, leaving protobuf 3.20 the only `3.x` version accepted by new TF. 

In other words, next TF release is expected to be protobuf 3.21.9 generation time (corresponds to 4.21.9 Python protobuf runtime) and be compatible with Python protobuf 3.20+.

**Comment by groodt** [comment]
@vam-google Amazing! Thanks!

Apologies for this question, Im sure you get it a lot. I tried to find it, but I couldn't see anything obvious. Is there an approximate ETA of tf 2.12 anywhere?

**Comment by nluehr** [comment]
@vam-google Thanks for the details on the protobuf upgrade. Given that this is already being handled elsewhere for the TF 2.12 release. I'm closing this PR.
